libcudacxx/include/cuda/std/__memory/assume_aligned.h

```diff
@@ -30,22 +31,24 @@
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
 template <size_t _Align, class _Tp>
-_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 _Tp* assume_aligned(_Tp* __ptr) noexcept
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr _Tp* assume_aligned(_Tp* __ptr) noexcept
```

Comment:
We do have a nontrivial amount of issues when it comes to nodiscard, so I would like to keep that as the macro
```suggestion
_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI constexpr _Tp* assume_aligned(_Tp* __ptr) noexcept
```